unity/Assets/Scripts/ARFlow/ARFlowClientManager.cs

```diff
@@ -0,0 +1,326 @@
+            }
+            if (Accelerometer.current != null)
+            {
+                InputSystem.EnableDevice(Accelerometer.current);
+            }
+            if (GravitySensor.current != null)
+            {
+                InputSystem.EnableDevice(GravitySensor.current);
+            }
+            _cameraManager = cameraManager;
+            _occlusionManager = occlusionManager;
+        }
+
+
+
+        /// <summary>
+        /// Connect to the server at an address, and with data modalities activated or not.
+        /// </summary>
+        /// <param name="address">Server address</param>
+        /// <param name="activatedDataModalities">Dictionary of all data modalities, either activated or not</param>
+        public void Connect(
+            string address,
+            Dictionary<string, bool> activatedDataModalities = null
+        )
+        {
+            _client = new ARFlowClient(address);
+            _activatedDataModalities = activatedDataModalities;
+            if (activatedDataModalities == null)
+                _activatedDataModalities = DEFAULT_MODALITIES;
+
+            try
+            {
+                _cameraManager.TryGetIntrinsics(out var k);
+                _cameraManager.TryAcquireLatestCpuImage(out var colorImage);
+                _occlusionManager.TryAcquireEnvironmentDepthCpuImage(out var depthImage);
+
+                _sampleSize = depthImage.dimensions;
+
+                var requestData = new RegisterRequest()
+                {
+                    DeviceName = SystemInfo.deviceName,
+                    CameraIntrinsics = new RegisterRequest.Types.CameraIntrinsics()
+                    {
+                        FocalLengthX = k.focalLength.x,
+                        FocalLengthY = k.focalLength.y,
+                        ResolutionX = k.resolution.x,
+                        ResolutionY = k.resolution.y,
+                        PrincipalPointX = k.principalPoint.x,
+                        PrincipalPointY = k.principalPoint.y,
+                    }
+
+                };
+                if (_activatedDataModalities["CameraColor"])
+                {
+                    var CameraColor = new RegisterRequest.Types.CameraColor()
+                    {
+                        Enabled = true,
+                        DataType = "YCbCr420",
+                        ResizeFactorX = depthImage.dimensions.x / (float)colorImage.dimensions.x,
+                        ResizeFactorY = depthImage.dimensions.y / (float)colorImage.dimensions.y,
+                    };
+                    requestData.CameraColor = CameraColor;
+                }
+                if (_activatedDataModalities["CameraDepth"])
+                {
+                    var CameraDepth = new RegisterRequest.Types.CameraDepth()
+                    {
+                        Enabled = true,
+#if UNITY_ANDROID
+                        DataType = "u16", // f32 for iOS, u16 for Android
+#endif
+#if UNITY_IPHONE
```

Comment:
Deprecated. Use UNITY_IOS instead. Check: https://docs.unity3d.com/Manual/PlatformDependentCompilation.html

Reply:
Good catch, thank you for the note.